pkg/sink/sink.go

```diff
@@ -72,27 +72,39 @@ func (r Resource) HandleEvent(response http.ResponseWriter, request *http.Reques
 	log.Printf("EventListener: %s in Namespace: %s handling event (EventID: %s) with payload: %s and header: %v",
 		r.EventListenerName, r.EventListenerNamespace, eventID, string(event), request.Header)
 
+	result := make(chan int, 10)
 	// Execute each Trigger
 	for _, trigger := range el.Spec.Triggers {
-		go r.executeTrigger(event, request, trigger, eventID)
+		go r.executeTrigger(event, request, trigger, eventID, result)
+	}
+
+	code := http.StatusAccepted
+	for i := 0; i < len(el.Spec.Triggers); i++ {
+		thiscode := <-result
+		if thiscode < code {
+			code = thiscode
+		}
```

Comment:
Please add a comment before this logic explaining the logic behind what return code is used (essentially repeat what you explained in the PR comment).

Reply:
I added the comment.